bot/exts/fun/off_topic_names.py

```diff
@@ -117,25 +152,134 @@ async def delete_command(self, ctx: Context, *, name: OffTopicName) -> None:
         log.info(f"{ctx.author} deleted the off-topic channel name '{name}'")
         await ctx.send(f":ok_hand: Removed `{name}` from the names list.")
 
-    @otname_group.command(name='list', aliases=('l',))
+    @otname_group.command(name='activate', aliases=('whitelist',))
+    @has_any_role(*MODERATION_ROLES)
+    async def activate_ot_name(self, ctx: Context, name: OffTopicName) -> None:
+        """Activate an existing off-topic name."""
+        await self.toggle_ot_name_activity(ctx, name, True)
+
+    @otname_group.command(name='deactivate', aliases=('blacklist',))
+    @has_any_role(*MODERATION_ROLES)
+    async def de_activate_ot_name(self, ctx: Context, name: OffTopicName) -> None:
+        """Deactivate a specific off-topic name."""
+        await self.toggle_ot_name_activity(ctx, name, False)
+
+    @otname_group.command(name='reroll')
+    @has_any_role(*MODERATION_ROLES)
+    async def re_roll_command(self, ctx: Context, ot_channel_index: Optional[int] = None) -> None:
+        """
+        Re-roll an off-topic name for a specific off-topic channel and deactivate the current name.
+
+        ot_channel_index: [0, 1, 2, ...]
+        """
+        if ot_channel_index is not None:
+            try:
+                channel = self.bot.get_channel(CHANNELS[ot_channel_index])
+            except IndexError:
+                await ctx.send(f":x: No off-topic channel found with index {ot_channel_index}.")
+                return
+        elif ctx.channel.id in CHANNELS:
+            channel = ctx.channel
+
+        else:
+            await ctx.send("Please specify channel for which the off-topic name should be re-rolled.")
+            return
+
+        old_channel_name = channel.name
+        old_ot_name = old_channel_name[NAME_START_INDEX:]  # ot1-name-of-ot -> name-of-ot
+
+        await self.de_activate_ot_name(ctx, old_ot_name)
+
+        response = await self.bot.api_client.get(
+            'bot/off-topic-channel-names', params={'random_items': 1}
+        )
+        try:
+            new_channel_name = response[0]
+        except IndexError:
+            await ctx.send("Out of active off-topic names. Add new names to reroll.")
+            return
+
+        async def rename_channel() -> None:
+            """Rename off-topic channel and log events."""
+            await channel.edit(
+                name=OTN_FORMATTER.format(number=old_channel_name[OT_NUMBER_INDEX], name=new_channel_name)
+            )
+            log.info(
+                f"{ctx.author} Off-topic channel re-named from `{old_ot_name}` "
+                f"to `{new_channel_name}`."
+            )
+
+            await ctx.message.reply(
+                f":ok_hand: Off-topic channel re-named from `{old_ot_name}` "
+                f"to `{new_channel_name}`. "
+            )
+
+        try:
+            await asyncio.wait_for(
+                rename_channel(),
+                3
+            )
```

Comment:
The formatting of this line is a little odd, shouldn't it fit on one line?
```suggestion
            await asyncio.wait_for(rename_channel(), 3)
```